### DISCUSSION TITLE
Interested in how people are visualising CVE's from the OSV option

### DISCUSSION THREAD

**Discussion by oc-christopher-billett** [renovatebot/renovate#31731] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We are using the OSV option to show CVE's within PR's. What we are interested in is how people are reporting or visualising these for compliance purposes. We run Grafana and have the ability to connect GitHub in but feel this wont necessarily give us the output we would need.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by HonkingGoose** [comment, marked as answer]
Maybe one of these items helps you?

- [Renovate docs, user stories, Swissquote](https://docs.renovatebot.com/user-stories/swissquote/) show a user-created dashboard to track Renovate PRs
- #23105
- #16624
- #23961
- https://github.com/renovatebot/renovate/discussions/26024#discussioncomment-7897831

**Reply by oc-christopher-billett** [reply]
Thank you i'll give it a read up

**Reply by oc-christopher-billett** [reply]
@HonkingGoose - For information, I was able to use the GitHub API to setup a dashboard that looks at PR's, their content and show the count of various CVSS Scores. I'll probably need to pull the rest of the data ineed into prometheus before I can make it into something usable

<img width="1047" alt="image" src="https://github.com/user-attachments/assets/7f71b54e-391f-4790-9d18-f8064a586cb2">

**Reply by HonkingGoose** [reply]
Thank you for letting me know you got it working, and for showing off your dashboard. :partying_face: 

Some ideas to improve your dashboard, in no particular order:

1. The green color for `Total Open Security Patches` is confusing me, I think it should either:
   - Match the most _severe_ color (so _yellow_ in your example screenshot)
   - Or use a neutral color, like white
2. There's a typo in the label for the `mediuem` score, the correct spelling is `medium`
3. Only capitalize the first word of each column
4. Only capitalize the start of the category name: `Low / Medium / High / Critical`
5. If possible, when you hover over the information icon, show a tooltip like: `The CVSS score range for this column is: x-z, based on the criteria of the CVSS organization.` This way it's easier to learn where the score comes from, and in what "bucket" the score lands.
6. Put the critical stuff near the _left_ of the table, instead of at the end. Like this:
    | Total open security patches | Critical score | High score | Medium score | Low score |
    | --- | --- | --- | --- | --- |
    | 1 | 0 | 0 | 1 | 0 |
